tests/unit/v1/test_async_vector_query.py

```diff
@@ -0,0 +1,241 @@
+
+    kwargs = make_retry_timeout_kwargs(retry=None, timeout=None)
+
+    # Execute the vector query and check the response.
+    firestore_api.run_query.return_value = AsyncIter([response_pb])
+
+    vector_query = collection_group_ref.where("snooze", "==", 10).find_nearest(
+        vector_field="embedding",
+        query_vector=Vector([1.0, 2.0, 3.0]),
+        distance_measure=distance_measure,
+        limit=5,
+    )
+
+    returned = await vector_query.get(transaction=_transaction(client), **kwargs)
+    assert isinstance(returned, list)
+    assert len(returned) == 1
+    assert returned[0].to_dict() == data
+
+    parent = client.collection("dee")
+    parent_path, expected_prefix = parent._parent_info()
+
+    expected_pb = _expected_pb(
+        parent=parent,
+        vector_field="embedding",
+        vector=Vector([1.0, 2.0, 3.0]),
+        distance_type=expected_distance,
+        limit=5,
+    )
+    expected_pb.where = StructuredQuery.Filter(
+        field_filter=StructuredQuery.FieldFilter(
+            field=StructuredQuery.FieldReference(field_path="snooze"),
+            op=StructuredQuery.FieldFilter.Operator.EQUAL,
+            value=encode_value(10),
+        )
+    )
+    expected_pb.from_ = [
+        StructuredQuery.CollectionSelector(collection_id="dee", all_descendants=True)
+    ]
+
+    firestore_api.run_query.assert_called_once_with(
+        request={
+            "parent": parent_path,
+            "structured_query": expected_pb,
+            "transaction": _TXN_ID,
+        },
+        metadata=client._rpc_metadata,
+        **kwargs,
+    )
+
+
+@pytest.mark.asyncio
+async def test_async_query_stream_multiple_empty_response_in_stream():
+    # Create a minimal fake GAPIC with a dummy response.
+    firestore_api = AsyncMock(spec=["run_query"])
+    empty_response1 = _make_query_response()
+    empty_response2 = _make_query_response()
+    run_query_response = AsyncIter([empty_response1, empty_response2])
+    firestore_api.run_query.return_value = run_query_response
+
+    # Attach the fake GAPIC to a real client.
+    client = make_async_client()
+    client._firestore_api_internal = firestore_api
+
+    # Make a **real** collection reference as parent.
+    parent = client.collection("dah", "dah", "dum")
+    async_vector_query = parent.where("snooze", "==", 10).find_nearest(
+        vector_field="embedding",
+        query_vector=Vector([1.0, 2.0, 3.0]),
+        distance_measure=DistanceMeasure.EUCLIDEAN,
+        limit=5,
+    )
+
+    result = [snapshot async for snapshot in async_vector_query.stream()]
+
+    assert list(result) == []
+
+    # Verify the mock call.
+    parent_path, _ = parent._parent_info()
+    firestore_api.run_query.assert_called_once_with(
+        request={
+            "parent": parent_path,
+            "structured_query": async_vector_query._to_protobuf(),
+            "transaction": None,
+        },
+        metadata=client._rpc_metadata,
+    )
```

Comment:
Could you add some system tests as well?

Reply:
SG. Done.